pkg/controller/factory/base_controller.go

```diff
@@ -156,6 +156,14 @@ func (c *baseController) runWorker(queueCtx context.Context) {
 	var workerWaitGroup sync.WaitGroup
 	workerWaitGroup.Add(1)
 	go func() {
+		defer utilruntime.HandleCrash(func(r interface{}) {
```

Comment:
correct spot, but I suggest separately adding to `runtime.PanicHandlers` as soon as you have the recorder in the command.